rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
You're right, though I suppose you mean `rcl` not `rclcpp`. You'll have to either remove the code and tests for the Python selection of rmw impl (as is proposed in this pr) or fix those functions/features to use the new mechanism, assuming that is possible.

I hadn't thought about it before, but that also means that you cannot build rmw implementation specific user executables at all anymore. Before I just assumed we were removing that logic because we could, not because we had too.

Doesn't it also mean that all of the `get_X_information` macros (where X is `rcl`, `rclcpp`, etc.) are no longer useful and should be removed? If that is the case, then that is another example of dead code that should be removed.